src/id128.rs

```diff
@@ -53,83 +44,130 @@ impl Id128 {
         // Set variant to DCE.
         hashed[8] = (hashed[8] & 0x3F) | 0x80;
 
-        Self::try_from_slice(&hashed[..16])
+        Self::try_from(&hashed[..16])
+    }
+
+    /// Return the unique boot ID.
+    pub fn from_boot() -> Result<Self, SdError> {
```

Comment:
As much as we may not like the original name, this is a port of `sd_id128_get_boot(3)` so it would be better to align the naming on that.

Reply:
Calling it `get_boot` would diverge from the Rust method naming scheme. It could just be called `boot` instead, but `from_boot` is what the [systemd crate](https://docs.rs/systemd/0.8.2/systemd/id128/struct.Id128.html#method.from_boot) named it.

Reply:
Indeed that's unfortunate, naming things is hard and aligning multiple beasts even harder. We cowardly refuse to walk into the minefield here and just go with that upstream has, see other comment for the rationale.